rest/src/main/scala/io/udash/rest/openapi/GeneratedSchemaName.scala

```diff
@@ -0,0 +1,33 @@
+package io.udash.rest.openapi
+
+import com.avsystem.commons.Opt
+import com.avsystem.commons.misc.{AnnotationOf, SimpleClassName}
+import com.avsystem.commons.serialization.name
+
+/**
+ * A metadata typeclass that you can use to control names of [[RestSchema]]s macro-materialized
+ * for ADTs (case classes & sealed hierarchies).
+ *
+ * [[GeneratedSchemaName]] is captured by [[RestStructure]] which is then used to construct
+ * a [[RestSchema]] instance.
+ *
+ * Customizing this implicit is mostly useful when used collectively, i.e. you can derive schema name
+ * as fully qualified class name from a `ClassTag`, in a generic `implicit def`.
+ *
+ * @param name the schema name or `Opt.Empty` for an unnamed (inline) schema
+ */
+final case class GeneratedSchemaName[T](name: Opt[String])
```

Comment:
Should it extend `com.avsystem.commons.meta.TypedMetadata`?

Reply:
No because it doesn't use `AdtMetadataCompanion` and all the associated complex macro-reflection associated with it.